fhirAssembly/build.gradle

```diff
@@ -0,0 +1,213 @@
+                    req.getOutputStream().write(sampleFile.bytes)
+                    logger.debug('Uploading to {}: {}', redirectedServer, new String(sampleFile.bytes))
+                    logger.info "Status code: ${req.getResponseCode()}" // HTTP request done on first read
+                    switch (req.getResponseCode()) {
+                        case HttpURLConnection.HTTP_MOVED_PERM:
+                        //noinspection FallThruinSwitch
+                        case HttpURLConnection.HTTP_MOVED_TEMP:
+                            def location = req.getHeaderField("Location")
+                            location = URLDecoder.decode(location, "UTF-8")
+                            def base = new URL(fhirServer)
+                            redirectedServer = new URL(base, location).toExternalForm()  // Deal with relative URLs
+                            logger.info('Redirected to {}', redirectedServer)
+                            redirects += 1
+                            break
+
+                            // Handle the set of 'it worked' conditions.  Most here should just be OK.
+                        case HttpURLConnection.HTTP_OK:
+                        //noinspection FallThruinSwitch
+                        case HttpURLConnection.HTTP_ACCEPTED:
+                        //noinspection FallThruinSwitch
+                        case HttpURLConnection.HTTP_CREATED:
+                            loaded += 1
+                            redirecting = false
+                            break
+
+                        default:
+                            errors += 1
+                            logger.error('HTTP Error loading {}: {}',
+                                sampleFile.name, req.getResponseCode())
+                            break
+                    }
+                    if (logger.isDebugEnabled()) {
+                        InputStream bis = new BufferedInputStream(req.getInputStream())
+                        String result = new String(bis.readAllBytes())
+                        logger.debug('Result: {}', result)
+                    }
+                }
+            } catch (Exception e) {
+                logger.error('Upload of {} failed: ', sampleFile.name, e)
+            }
+        }
+        logger.quiet('uploadSampleData: {} attempts, {} successes, {} errors, {} redirects',
+            attempts, loaded, errors, redirects)
+    }
+}
+
+// This defines an in-memory FHIR server.  The H2 database, sans any configuration, runs in memory only.  So any
+// changes that happen as a result of the tests are lost on shutdown.  We'll reload the sample data on each startup
+// (see task dependencies below with uploadSampleData).
+dcompose {
+    namePrefix = 'test_'
+    fhir {
+        image = 'hapiproject/hapi:latest'
+        exposedPorts = [ '8080' ]
+        portBindings = ['8090:8080']
+        env = [
+            'HAPI_FHIR_SERVER_ADDRESS=http://localhost:8090/fhir'
+        ]
+    }
+}
+
+/**
```

Comment:
The server takes a while to start.  So, we'll repeatedly try & fetch the metadata/conformance statement.  This isn't available until to server is ready to go, so it suffices as a a ready check.  We have a configurable max time to wait, & we try to sample it periodically without sucking up the whole machine.  Generally, it seems to take 20-50 seconds.